thesis-de/content-de.tex

```diff
@@ -951,7 +951,45 @@ \subsubsection{Durchführung der Evaluierung}
 
 \paragraph{Cassandra}
 \label{paragraph:cassandra}
-Some Text here!
+Cassandra, ist ein \gls{SQL} und Java basiertes \gls{DBMS} und wurde ausgelegt
+für die Datenhaltung von großen Datenmengen und dessen verbundenen
+Management der Arbeitsauslastung auf dem Server. Dazu nutzt die Software ein
+Cluster System über mehrere Server (Nodes) um ein vollständiges Versagen, durch
+den Ausfall von Hardwarekomponenten oder Softwareteilen auf dem System zu
+verhindern. Dazu wird ein Peer-to-Peer Verfahren verwendet, welches die
+Dateninformationen auf mehrere Server verteilt und verwaltet. Im Cassandra
+Umfeld wird dies über die von dem Hersteller definierten Protokoll ``gossip''
+verwaltet. Es handelt sich dabei um ein Kommunikationsprotokoll, welches den
+Austausch von Statusinformationen und Metainformationen zwischen den einzelnen
+Nodes ermöglicht. Wenn auf einem einzelnen Node Dateninformationen erstellt,
+modifiziert oder entfernt werden, erhält ein weiterer Node in der Infrastruktur
+diese Information und baut gleichzeitig eine Replikation des Objektes. 
+
+\lstinputlisting[language=bash,caption={postgres git clone}]{../listings/cassandra-status.txt}
+
+In der obigen Shellausgabe wird ein während der Evaluierungsphase erstelltes
+Cassandra System dargestellt. Dabei besitzt der Prototyp zwei aktive Nodes,
+welche für Testzwecke auf dem gleichen physischen Server betrieben werden. Im
+Wirkbetrieb sollten diese auf zwei verschiedenen Serversystemen arbeiten, um
+die Ausfallsicherheit der Hardware zu gewährleisten. Es ist jedoch zu erkennen,
+dass beide Systeme einen Dateneigentum, in der Shellausgabe die Spalte
+``Owns'', von 50 Prozentpunkten besitzen. Dies bedeutet, dass jeder Node
+jeweils die Hälfte der Dateninformationen bereitstellt, jedoch auch die
+zugehörige Hälfte von dem anderen Node als Backup abspeichert. Sollte nun
+einer der beiden Nodes ausfallen, könnte der jeweils andere automatisch
+einen Dateneigentum von 100 Prozentpunkten erstellen und ohne Verluste
+weiterarbeiten.
```

Comment:
kannst du noch das stichwort Shared-Nothing-Architektur einbauen?
die bedeutet, das alles nodes im cluster den dienst bereitstellen können und nicht zwangsläufig auf andere nodes angewiesen sind, dies is hier ja der fall aufgrund der replika. Ich nutze das wort schon irgendwo weiter oben.